### PR TITLE
Test suite

### PR DESCRIPTION
**Pull request by gkellogg**
* Initial tests.
* Reference tests from spec.
* Build HTML test manifests with cross-reference back to the spec.
* GH Action to build HTML versions of test manifests and push back to branch. (removed for now).

For #20.

***
<a href="https://pr-preview.s3.amazonaws.com/json-ld/yaml-ld/pull/86.html" title="Last updated on Sep 30, 2022, 5:20 PM UTC (d50b7a6)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/json-ld/yaml-ld/86/1976cf3...d50b7a6.html" title="Last updated on Sep 30, 2022, 5:20 PM UTC (d50b7a6)">Diff</a>

### CODE REVIEW

**Comment by gkellogg** [comment]
Despite the successful run, the action which tries to push up the built manifest files is not working. Need more time to work on this, and if anyone has more experience doing this kind of thing, please jump right in.

File: tests/README.md
Comment (TallTed) [inline tests/README.md:64]:
```suggestion
The HTML versions of the test manifests are built using the Node action `npm run generate`, which also cross-indexes tests with references from the specification.
```

File: tests/README.md
Comment (TallTed) [inline tests/README.md:20]:
```suggestion
The _expected_ results with file extension `.jsonld` or `.yamlld` can be compared using [YAML-LD object comparison](#yaml-ld-object-comparison) with the processor output. Additionally, if the `ordered` option is not set, the result should be expanded and compared with the expanded _expected_ document also using [YAML-LD object comparison](#yaml-ld-object-comparison).
```

File: tests/README.md
Comment (TallTed) [inline tests/README.md:24]:
```suggestion
Tests of type `jld:ToRDFTest` have an _expected_ file extension of `.nq`. Their results can be compared using [RDF Dataset Isomorphism](https://www.w3.org/TR/rdf11-concepts/#dfn-dataset-isomorphism).
```

File: tests/README.md
Comment (TallTed) [inline tests/README.md:36]:
```suggestion
YAML-LD object comparison recursively compares JSON objects, arrays, and values for equality.
```

File: tests/README.md
Comment (TallTed) [inline tests/README.md:39]:
```suggestion
* JSON arrays are generally compared without regard to order (the lone exception being if the referencing key is `@list`). Each item within the array must be equivalent to an item in the array being compared to by recursively using the comparison algorithm. For values of `@list`, the order of these items is significant.
```

File: tests/README.md
Comment (TallTed) [inline tests/README.md:41]:
```suggestion
* Values of `@language`, and other places where language tags may be used, are specified in lowercase in the test results. Implementations should either normalize language tags for testing purposes, or compare language tags in a case-independent way.
```

File: tests/README.md
Comment (TallTed) [inline tests/README.md:58]:
```suggestion
* Some _toRdf_ tests may generate results as [Generalized RDF](https://www.w3.org/TR/rdf11-concepts/#section-generalized-rdf), specifically having a blank node predicate. Technically, these are in an invalid N-Quads format; provision must be made to parse and recognize quads having a blank node predicate.
```

File: tests/README.md
Comment (TallTed) [inline tests/README.md:71]: Though I do wonder, why not suggest they create an issue on the repo?
```suggestion
1. Notify the JSON-LD Working Group mailing list, [public-linked-json@w3.org](mailto:public-linked-json@w3.org),
```

File: tests/basic-manifest.jsonld
Comment (pchampin) [inline tests/basic-manifest.jsonld:20]: I guess this is meant to be "UTF-16"?

File: tests/README.md
Comment (pchampin) [reply tests/README.md:71]: +1 to also add a reference to the github repo

**Review by pchampin — APPROVED** [review]
LGTM modulo minor remarks

File: tests/basic-manifest.jsonld
Comment (TallTed) [reply tests/basic-manifest.jsonld:20]: See also apparently related line 136 —
```
                Test cr-utf8-negative-1 UTF-6 BOM
```
— and line 142 —
```
                  <dt>Name</dt><dd>UTF-6 BOM</dd></dt>
```
— of tests/basic-manifest.html

Comment (gkellogg) [reply tests/basic-manifest.jsonld:20]: Those last two are derived from the .jsonld version.

File: tests/README.md
Comment (gkellogg) [reply tests/README.md]: Check updated document, also in generated tests/*manifest.html files.

**Comment by gkellogg** [comment]
A couple of more changes, including comments on generated files that changes should be made to the source.

I did try to make these files ignored and have them built by GH Action, but failed in that. I notice in the ePub tests, generated files are only made to the main branch when a PR is merged, which may ultimately be the way to go. @iherman may have some thoughts on this.

File: tests/README.md
Comment (TallTed) [inline tests/README.md:74]:
```suggestion
  or by sending a message to the group mailing list, [public-linked-json@w3.org](mailto:public-linked-json@w3.org).
```

**Comment by gkellogg** [comment]
Thanks for the improvements, I think it will be more useful for people once this is merged. I'm working on additional tests that will come in another PR.###

**Pull request merged**
